### PR TITLE
feat: Enable setting entire store state

### PR DESCRIPTION
**Pull request by teamchong**
Previously, `oauthStore.state = { codeVerifier, state, location };` is not possible. Setting the entire state with an object is not allowed. This limitation hindered the flexibility of state management and made it difficult to override the entire state of the proxy.

To resolve this, we've added a setter to the store that allows passing an entire object. This enhancement enables overriding the entire state of the proxy with a single operation.

Now, the store state can be easily manipulated and overridden as needed, improving the flexibility and ease of state management.

Closes [UI-96](https://linear.app/eventstore/issue/UI-89/allow-setting-entire-store-state).

### CODE REVIEW

File: packages/stores/src/stores/createStore.ts [hunk 1]
```diff
@@ -2,10 +2,35 @@ import type { OnHandler, OnChangeHandler, Handlers, Plugin } from '../types';
...
+                              | triggers         | triggered by
++-----------------+           |                  |
+|       use       | <----------------------------+
+| (Registers      |                              
+|   plugins and   |                     +-----------------+
+|   ties plugin   |                     |     Plugins     |
+|   methods to    |-------------------> | (Extend Store's |
+|       handlers) |      registers      |  functionality) |   
++-----------------+                     +-----------------+  */
+
```
Comment (George-Payne) [inline packages/stores/src/stores/createStore.ts:29]: :+1:

File: packages/stores/src/stores/createStore.ts [hunk 2]
```diff
@@ -175,7 +200,30 @@ export const createStore = <T extends { [key: string]: any }>(
...
 
     const store: Store<T> = {
-        state,
+        get state() {
+            return state;
+        },
+        set state(value) {
+            if (value === null || typeof value !== 'object') {
+                throw new TypeError(
+                    `Expected object, null, or undefined, got ${typeof value}`,
```
Comment (George-Payne) [inline packages/stores/src/stores/createStore.ts:209]:
```suggestion
                    `Expected object, got ${typeof value}`,
```

You're not excepting `null` or `undefined` here

File: packages/stores/src/stores/createStore.test.ts [hunk 3]
```diff
@@ -504,3 +504,58 @@ describe('use subscriptions', () => {
...
+            store.state = (undefined as unknown) as State;
+        }).toThrow(TypeError);
+        expect(() => {
+            store.state = (null as unknown) as State;
+        }).toThrow(TypeError);
+        expect(() => {
+            store.state = 'hello world' as State;
+        }).toThrow(TypeError);
+    });
+});
```
Comment (George-Payne) [inline packages/stores/src/stores/createStore.test.ts:561]: Probably worth adding a test that `on` triggers for the change,

**Review by George-Payne — CHANGES_REQUESTED** [review]
Nice work, just a couple more tests

File: packages/stores/src/stores/createStore.test.ts (hunk 3 above)
Comment (teamchong) [reply packages/stores/src/stores/createStore.test.ts:589]: Added test for state change triggers.

**Review by George-Payne — APPROVED** [review]

**Pull request merged**